modules/local/gene_summary/templates/gene_summary.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env python3
+
+# versions
+import platform
+import yaml
+
+import os
+os.environ["MPLCONFIGDIR"] = "./tmp/mpl"
+os.environ["NUMBA_CACHE_DIR"] = "./tmp/numba"
+
+import pandas as pd
+import scanpy as sc
+import numpy as np
+import mudata as md
+import pegasusio as io
+
+from pathlib import Path
+from mudata import MuData
+from typing import Tuple
+
+class Arguments:
+    # adopted from mygene module (Suzanne Jin)
+    """
+    Parses the arguments, including the ones coming from $task.ext.args.
+    """
+
+    def __init__(self) -> None:
+
+        self.singlet_str = "singlet"
+        self.doublet_str = "doublet"
+        self.negative_str = "negative"
+        self.parse_input_args()
+        self.creat_output_dirs()
+        #self.testing_inputs()
+
+    def parse_input_args(self) -> None:
+
+        self.prefix = "$task.ext.prefix" if "$task.ext.prefix" != "null" else "$meta.id"
+
+        self.rna_matrix               = "${rna_matrix}"
+        self.hto_matrix               = "${hto_matrix}"
+        self.barcodes                 = "${barcodes}"
+        self.vireo                    = "${vireo}"
+        self.demuxlet                 = "${demuxlet}"
+        self.freemuxlet               = "${freemuxlet}"
+        self.souporcell               = "${souporcell}"
+
+        self.generate_anndata         = "${generate_anndata}"
+        self.generate_mudata          = "${generate_mudata}"
+
+        path_vars = {
+            "rna_matrix",
+            "hto_matrix",
+            "barcodes",
+            "vireo",
+            "demuxlet",
+            "freemuxlet",
+            "souporcell"
+        }
+
+        boolean_vars = {
+            "generate_anndata",
+            "generate_mudata"
+        }
+
+        def _tranlate_to_python(input_str,value_str):
```

Comment:
```suggestion
        def _translate_to_python(input_str: str, value_str: str) -> bool:
```

The whole file seems badly formatted. Can you run `ruff-check` and `ruff-format` on the python scripts, please?

Reply:
```
ruff --version 
ruff 0.12.1

ruff check modules/local/gene_summary/templates/gene_summary.py
All checks passed!

ruff format modules/local/gene_summary/templates/gene_summary.py
1 file reformatted
```

Thank you, ruff format  helped!